ext/drx/scatter_gather_aarch64.c

```diff
@@ -641,11 +805,19 @@ drx_expand_scatter_gather(void *drcontext, instrlist_t *bb, OUT bool *expanded)
     /* Tell drx_event_restore_state() that an expansion has occurred. */
     drx_mark_scatter_gather_expanded();
 
-    reg_id_t scratch_gpr0 = DR_REG_INVALID;
-    reg_id_t scratch_gpr1 = DR_REG_INVALID;
+#define MAX_SCRATCH_GPR 3
+    reg_id_t scratch_gpr[MAX_SCRATCH_GPR] = { DR_REG_INVALID, DR_REG_INVALID,
+                                              DR_REG_INVALID };
     drvector_t allowed;
     drreg_init_and_fill_vector(&allowed, true);
 
+    /* Figure out how many scratch registers we need */
+    uint num_scratch_gpr_needed = 1;
+    if (!sg_info.is_load)
+        num_scratch_gpr_needed++;
+    if (is_contiguous)
+        num_scratch_gpr_needed++;
```

Comment:
Would be nice to add a comment over all three of these to say exactly what app reg we need the scratch for.

E.g. here we can say that it's for the gpr that we use to store the base reg of the contiguous memory range.

Reply:
Done.